File: app/src/main/java/com/starostinvlad/rxeducation/DatumDao.java
package com.starostinvlad.rxeducation;

import com.starostinvlad.rxeducation.GsonModels.Datum;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface DatumDao {

    @Query("SELECT * FROM datum")
    List<Datum> getAll();

    @Query("SELECT * FROM datum WHERE id = :id")
    Datum getById(long id);

    @Insert
    void insert(List<Datum> datum);

    @Insert
    void insert(Datum datum);

    @Update
    void update(Datum datum);

    @Delete
    void delete(Datum datum);

}

File: app/src/main/java/com/starostinvlad/fan/DatumDao.kt
package com.starostinvlad.fan

import androidx.room.*
import com.starostinvlad.fan.GsonModels.Datum
import kotlin.Throws

@Dao
interface DatumDao {
    @get:Query("SELECT * FROM datum")
    val all: List<Datum?>?

    @Query("SELECT * FROM datum WHERE id = :id")
    fun getById(id: Long): Datum?

    @Insert
    fun insert(datum: List<Datum?>?)

    @Insert
    fun insert(datum: Datum?)

    @Update
    fun update(datum: Datum?)

    @Delete
    fun delete(datum: Datum?)
}